test/functional/sql-tag/sql-tag-parameters-mysql.ts

```diff
@@ -0,0 +1,189 @@
+                expect(ids).to.have.members(["null1"])
+            }),
+        ))
+
+    it("should handle SQL tag parameters with boolean values", () =>
+        Promise.all(
+            connections.map(async (connection) => {
+                const repo = connection.getRepository(MysqlExample)
+
+                await repo.save([
+                    { id: "true1", active: true },
+                    { id: "false1", active: false },
+                ])
+
+                const value = true
+
+                const examples = await connection.sql`
+                    SELECT * FROM example WHERE active = ${value}
+                `
+
+                const ids = examples.map((e: MysqlExample) => e.id)
+
+                expect(examples).to.have.length(1)
+                expect(ids).to.have.members(["true1"])
+            }),
+        ))
+
+    it("should handle SQL tag parameters with date values", () =>
+        Promise.all(
+            connections.map(async (connection) => {
+                const repo = connection.getRepository(MysqlExample)
+                const now = new Date()
+                const yesterday = new Date(now.getTime() - 24 * 60 * 60 * 1000)
+                const nowStr = toMySQLDateString(now)
+                const yesterdayStr = toMySQLDateString(yesterday)
+
+                await repo.save([
+                    { id: "today", createdAt: nowStr },
+                    { id: "yesterday", createdAt: yesterdayStr },
+                ])
+
+                const examples = await connection.sql`
+                    SELECT * FROM example WHERE createdAt > ${yesterdayStr}
+                `
+
+                const ids = examples.map((e: MysqlExample) => e.id)
+
+                expect(examples).to.have.length(1)
+                expect(ids).to.have.members(["today"])
+            }),
+        ))
+
+    it("should handle SQL tag parameters with array values", () =>
+        Promise.all(
+            connections.map(async (connection) => {
+                const repo = connection.getRepository(MysqlExample)
+
+                await repo.save([
+                    { id: "array1", tags: "tag1,tag2" },
+                    { id: "array2", tags: "tag3,tag4" },
+                    { id: "array3", tags: "tag5,tag6" },
+                ])
+
+                const searchTags = ["tag1", "tag3"]
+
+                // Use FIND_IN_SET for MySQL to check if any tag matches
+                const examples = await connection.sql`
+                    SELECT * FROM example
+                    WHERE (
+                        FIND_IN_SET(${searchTags[0]}, tags) > 0
+                        OR FIND_IN_SET(${searchTags[1]}, tags) > 0
+                    )
+                `
+
+                const ids = examples.map((e: MysqlExample) => e.id)
+
+                expect(examples).to.have.length(2)
+                expect(ids).to.have.members(["array1", "array2"])
+            }),
+        ))
```

Comment:
This test doesn't prove anything since it's not actually binding an array value to the parameter. It should be replaced with a test that asserts the expected behavior of array parameters.

Reply:
You're right. I ended up doing a simpler query since the tests are running on MySQL 5.7 which lacks many features involved around arrays (8.0+)